modin/utils.py

```diff
@@ -58,15 +64,143 @@ def _get_indent(doc: str) -> int:
     int
         Minimal indent (excluding empty lines).
     """
+    indents = _get_indents(doc)
+    return min(indents) if indents else 0
+
+
+def _get_indents(source: Union[list, str]) -> list:
+    """
+    Compute indentation for each line of the source string.
+
+    Parameters
+    ----------
+    source : str or list of str
+        String to compute indents for. Passed list considered
+        as a list of lines of the source string.
+
+    Returns
+    -------
+    list of ints
+        List containing computed indents for each line.
+    """
     indents = []
-    for line in doc.splitlines():
+
+    if not isinstance(source, list):
+        source = source.splitlines()
+
+    for line in source:
         if not line.strip():
             continue
         for pos, ch in enumerate(line):
             if ch != " ":
                 break
         indents.append(pos)
-    return min(indents) if indents else 0
+    return indents
+
+
+def format_string(template: str, **kwargs) -> str:
```

Comment:
this function better have a test...

Reply:
it already has [one](https://github.com/modin-project/modin/pull/3021/files#diff-003a411f9eba9c58275576d3bfeddd0815d72660fce91c34f77052d6fe5b92a5R185)